File: ValidAnagram.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
    	
    	if(s.length() != t.length()) return false;
        
        vector<int> map(26, 0);
        
        for(int i = 0;i < s.length();i++) {
        	map[s[i] - 'a'] ++;
		}
		
		for(int i = 0;i < t.length();i++) {
			if(map[t[i] - 'a'] == 0) return false;
			map[t[i] - 'a'] --;
		}
		
		return true;
        
    }
};

int main() {
	
	string s, t;
	cin>>s;
	cin>>t;
	
	Solution *solution = new Solution();
	solution->isAnagram(s, t) ? cout<<"True" : cout<<"False";
	
	return 0;
	
}

File: ValidAnagram.py
class Solution:
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        dic = {}
        for c in s:
            if c not in dic:
                dic[c] = 0
            dic[c] += 1
        for c in t:
            if c not in dic:
                return False
            if dic[c] == 0:
                return False
            dic[c] -= 1
        return True

if __name__ == '__main__':
    s = input()
    t = input()
    solution = Solution()
    print(solution.isAnagram(s, t))
